### PR TITLE
feat: add configuration api route

### LINKED ISSUE #449: [Feature]: Support remote cache through API
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [K8sGPT Slack](https://k8sgpt.slack.com) and got positive feedback

### Is this feature request related to a problem?

No

### Problem Description

_No response_

### Solution Description

Relates to https://github.com/k8sgpt-ai/k8sgpt-operator/issues/113 and supporting the remote cache through the operator - we need a way to program it on an analyze request.

### Benefits

N/A

### Potential Drawbacks

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by matthisholleville**
Closes #449 

## 📑 Description

Initialize the configuration API route and add cache support

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

## ℹ Additional Information

Examples : 

`AddCache` : 

```bash
grpcurl -plaintext -d '{
  "cache": {
    "region": "us-west-1",
    "bucketName": "my-cache-bucket"
  }
}' localhost:8080 schema.v1.Server/AddConfig

{
  "status": "Configuration updated."
}
```

`RemoveCache` :

```bash
grpcurl -plaintext -d '{
  "cache": {
    "bucketName": "my-cache-bucket"
  }
}' localhost:8080 schema.v1.Server/RemoveConfig

{
  "status": "Successfully removed the remote cache"
}
```

### CODE REVIEW

**Comment by AlexsJones** [comment]
```
k8sgpt on  feat/init-cache-api [$] via 🐹 v1.20.4 on ☁️  (eu-west-1)
❯ grpcurl -plaintext -d '{
  "cache": {
    "bucketName": "foobark8sgpt1"
  }
}' localhost:8080 schema.v1.Server/RemoveConfig

Error invoking method "schema.v1.Server/RemoveConfig": target server does not expose service "schema.v1.Server"
```

@matthisholleville I tried this by checking your PR out, anything I am doing wrong here?

File: pkg/cache/cache.go [hunk 1]
```diff
@@ -39,3 +41,43 @@ func RemoteCacheEnabled() (bool, error) {
 	}
 	return false, nil
 }
+
+func AddCache(bucketName string, region string) error {
```
Comment (AlexsJones) [inline pkg/cache/cache.go:45]: Can we rename this as technically this is only for adding a remote S3 cache?

Maybe `AddRemoteS3Cache` or `AddRemoteCache` ?

File: pkg/cache/cache.go [hunk 2]
```diff
@@ -39,3 +41,43 @@ func RemoteCacheEnabled() (bool, error) {
...
+	cacheInfo.Region = region
+	viper.Set("cache", cacheInfo)
+	err = viper.WriteConfig()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func RemoveCache(bucketName string) error {
```
Comment (AlexsJones) [inline pkg/cache/cache.go:64]: Same as above, just to be really clear

File: pkg/server/server.go [hunk 3]
```diff
@@ -69,7 +69,7 @@ func (s *Config) Serve() error {
 	grpcServerUnaryInterceptor := grpc.UnaryInterceptor(logInterceptor(s.Logger))
 	grpcServer := grpc.NewServer(grpcServerUnaryInterceptor)
 	reflection.Register(grpcServer)
-	rpc.RegisterServerServer(grpcServer, s.Handler)
+	rpc.RegisterServerServiceServer(grpcServer, s.Handler)
```
Comment (AlexsJones) [inline pkg/server/server.go:72]: That's a rather long name, is that because `Server` gets suffixed?

**Comment by matthisholleville** [comment]
Try this @AlexsJones  : 

```bash
grpcurl -plaintext -d '{
  "cache": {
    "bucketName": "foobark8sgpt1"
  }
}' localhost:8080 schema.v1.ServerService/RemoveConfig
```

----

```bash
grpcurl -plaintext -d '{
  "cache": {
    "region": "us-west-1",
    "bucketName": "my-cache-bucket"
  }
}' localhost:8080 schema.v1.ServerService/AddConfig
```

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**